docs/sphinx/api/default_ops.rst

```diff
@@ -541,4 +541,74 @@ This is a non-linear transformation, and no template overloads are available.
     .. code-block:: cpp
 
         cudaq::qubit qubit;
-        my(qubit);
\ No newline at end of file
+        my(qubit);
+
+
+User-defined Custom Operations
+==============================
+
+Users can define a custom quantum operation by its unitary matrix. First use 
+the API to register a custom operation, outside of a CUDA-Q kernel. Then the 
+operation can be used within a CUDA-Q kernel like any of the built-in operations
+defined above.
+Custom operations are supported on qubits only (`qudit` with `level = 2`). For 
+multi-qubit operations, the matrix is interpreted with MSB qubit ordering.
```

Comment:
I'm not sure that everyone will interpret "MSB qubit ordering" the same way unless we provide a more explicit example. I think that in addition to showing what you have that also showing a more sophisticated example like a combined "XY" gate on 2 qubits (X on qubit 0 and Y on qubit 1) would be helpful. The result is not symmetrical, so it should help users get all of the ordering correct.

Reply:
I added an example to define CNOT operation to clarify this. See cc43fff33919ba6d1aa98985efc8e71863f48e62

Reply:
Perhaps we should add multiple examples? Can you help me construct this one to convey this...

Reply:
Yes, will do. It would be a good reason for me to experiment with it anyway 😄

Reply:
OK, here are some draft updates: https://github.com/NVIDIA/cuda-quantum/commit/3bae23e30a944784c8b7c62ecc56b6638d113e6f. If you like them, feel free to cherry pick that commit into your branch.

On a separate note, we should probably also put the code snippets into the `docs/sphinx/snippets` folder and reference them with `.. literalinclude` similar to how that is handled here: https://github.com/NVIDIA/cuda-quantum/blob/ad367c1bc562a11255fce11a0cecebe88a455d7b/docs/sphinx/using/quick_start.rst?plain=1#L78

The CI runs items in the snippets directory, so if the syntax ever changes, this will ensure that we keep the corresponding docs up to date.

Reply:
Addressed in beecd2aa42f953df3c02b1f0ff2f0414af538343